### PR TITLE
Add custom Python 3 Encoder for bytes and sets

### PR DESCRIPTION
**Pull request by dvvanessastoiber**
Fixes issues in Jamborees

### Summary
* the encoder adds `list()` to instances of type `set()` and therefore provides compatibility with Python 3.7
* furthermore, it changes the type `bytes` to `string`

### CODE REVIEW

**Comment by dvvanessastoiber** [comment]
Possibility to test functionality:

```
@app.route('/test', methods=['GET'])
def test():
  return jsonify(set())
```

in _init.py_:
```
registry.append('json-encoder', 'python3-encoder', 'phovea_server.python3_encoder', {})
```


After creating an empty `set()` and calling the respective route, the result is either an Exception or - in case of the encoder working correctly - an empty set `[]`

**Comment by puehringer** [comment]
@dvvanessastoiber It might be better to separate the concerns here: 
Rename the encoder to `SetEncoder` and **only** handle the `sets`, but not the `bytes` (which is handled in `tdp_core` by the `BytesToStringEncoder`). 
Also, add it to the `__init__.py` directly here in, because this encoder will be needed everywhere and not just Jamborees (the Jamborees PR can be deleted then).

**Comment by dvvanessastoiber** [comment]
> @dvvanessastoiber It might be better to separate the concerns here:
> Rename the encoder to `SetEncoder` and **only** handle the `sets`, but not the `bytes` (which is handled in `tdp_core` by the `BytesToStringEncoder`).
> Also, add it to the `__init__.py` directly here in, because this encoder will be needed everywhere and not just Jamborees (the Jamborees PR can be deleted then).

You are right, thank you!

File: phovea_server/set_encoder.py [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
 
 
-class Python3Encoder(object):
+class SetEncoder(object):
   def __contains__(self, obj):
-    if isinstance(obj, bytes):
-      return True
     if isinstance(obj, set):
       return True
     return False
```
Comment (puehringer) [inline phovea_server/set_encoder.py]: Can be changed to `return isinstance(obj, set)`.

File: phovea_server/set_encoder.py [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
 
   def __call__(self, obj, base_encoder):
-    if isinstance(obj, bytes):
-      return obj.decode('utf-8')
     if isinstance(obj, set):
       return list(obj)
     return None
```
Comment (puehringer) [inline phovea_server/set_encoder.py]: Due to the way this Encoder is called (https://github.com/phovea/phovea_server/blob/develop/phovea_server/util.py#L24-L26), these lines are only called on actual `set` instances. Therefore, they cannot be `None` or any other instance. You can replace this by `return list(obj)` then.

**Comment by anita-steiner** [comment]
please add the encoder to the yo-rc.json extensions

**Comment by dvvanessastoiber** [comment]
> please add the encoder to the yo-rc.json extensions

See https://github.com/phovea/phovea_server/pull/108

**Pull request merged**